lib/govuk_app_config/govuk_error/configure_defaults.rb

```diff
@@ -0,0 +1,82 @@
+require "govuk_app_config/govuk_error/govuk_data_sync"
+
+module GovukError
+  class ConfigureDefaults < SimpleDelegator
+    attr_accessor :data_sync_excluded_exceptions
+
+    def initialize(config)
+      super
+
+      config.current_environment = ENV["SENTRY_CURRENT_ENV"]
+
+      config.before_send = proc { |e|
+        GovukStatsd.increment("errors_occurred")
+        GovukStatsd.increment("error_types.#{e.class.name.demodulize.underscore}")
+        e
+      }
+
+      config.silence_ready = !Rails.env.production? if Object.const_defined?("Rails")
+
+      config.excluded_exceptions = [
+        # Default ActionDispatch rescue responses
+        "ActionController::RoutingError",
+        "AbstractController::ActionNotFound",
+        "ActionController::MethodNotAllowed",
+        "ActionController::UnknownHttpMethod",
+        "ActionController::NotImplemented",
+        "ActionController::UnknownFormat",
+        "Mime::Type::InvalidMimeType",
+        "ActionController::MissingExactTemplate",
+        "ActionController::InvalidAuthenticityToken",
+        "ActionController::InvalidCrossOriginRequest",
+        "ActionDispatch::Http::Parameters::ParseError",
+        "ActionController::BadRequest",
+        "ActionController::ParameterMissing",
+        "Rack::QueryParser::ParameterTypeError",
+        "Rack::QueryParser::InvalidParameterError",
+        # Default ActiveRecord rescue responses
+        "ActiveRecord::RecordNotFound",
+        "ActiveRecord::StaleObjectError",
+        "ActiveRecord::RecordInvalid",
+        "ActiveRecord::RecordNotSaved",
+        # Additional items
+        "ActiveJob::DeserializationError",
+        "CGI::Session::CookieStore::TamperedWithCookie",
+        "GdsApi::HTTPIntermittentServerError",
+        "GdsApi::TimedOutException",
+        "Mongoid::Errors::DocumentNotFound",
+        "Sinatra::NotFound",
+      ]
+
+      # This will exclude exceptions that are triggered by one of the ignored
+      # exceptions. For example, when any exception occurs in a template,
+      # Rails will raise a ActionView::Template::Error, instead of the original error.
+      config.inspect_exception_causes_for_exclusion = true
+
+      config.transport_failure_callback = proc {
+        GovukStatsd.increment("error_reports_failed")
+      }
+
+      data_sync = GovukDataSync.new(ENV["GOVUK_DATA_SYNC_PERIOD"])
+      @data_sync_excluded_exceptions = ["PG::Error"]
+      config.should_capture = lambda do |error_or_event|
+        exception_chain = Raven::Utils::ExceptionCauseChain.exception_to_array(error_or_event)
+        data_sync_ignored_error = false
+        data_sync_excluded_exceptions.each do |exception_to_ignore|
+          break if data_sync_ignored_error
+
+          data_sync_ignored_error = exception_chain.reduce(false) do |memo, exception|
+            memo || exception.class == exception_to_ignore ||
```

Comment:
I think we've got an issue here where exception.class will be a class object whereas exception_to_ignore will be a string?

Reply:
In my tests it would still pass whether or not a `.to_s` is appended to `.class`. I think the class equality method does some automatic conversion to a string if that's what it's being compared to. Perhaps a `.to_s` would reduce confusion, even if it's slightly verbose. I'll add it in.